src/PathTracedVolume.ts

```diff
@@ -132,148 +110,24 @@ export default class PathTracedVolume implements VolumeRenderImpl {
-    };
-
-    this.screenOutputShader = {
-      uniforms: UniformsUtils.merge([
-        {
-          gInvExposure: {
-            type: "f",
-            value: 1.0 / (1.0 - 0.75),
-          },
-          tTexture0: {
-            type: "t",
-            value: null,
-          },
-        },
-      ]),
-
-      vertexShader: [
-        "precision highp float;",
-        "precision highp int;",
-
-        "out vec2 vUv;",
-
-        "void main()",
-        "{",
-        "vUv = uv;",
-        "gl_Position = vec4( position, 1.0 );",
-        "}",
-      ].join("\n"),
-
-      fragmentShader: [
-        "precision highp float;",
-        "precision highp int;",
-        "precision highp sampler2D;",
-
-        "uniform float gInvExposure;",
-        "uniform sampler2D tTexture0;",
-        "in vec2 vUv;",
-
-        // Used to convert from XYZ to linear RGB space
-        "const mat3 XYZ_2_RGB = (mat3(",
-        "  3.2404542, -1.5371385, -0.4985314,",
-        " -0.9692660,  1.8760108,  0.0415560,",
-        "  0.0556434, -0.2040259,  1.0572252",
-        "));",
-
-        "vec3 XYZtoRGB(vec3 xyz) {",
-        "return xyz * XYZ_2_RGB;",
-        "}",
-
-        "void main()",
-        "{",
-        "vec4 pixelColor = texture(tTexture0, vUv);",
-
-        "pixelColor.rgb = XYZtoRGB(pixelColor.rgb);",
-
-        //'pixelColor.rgb = pow(pixelColor.rgb, vec3(1.0/2.2));',
-        "pixelColor.rgb = 1.0-exp(-pixelColor.rgb*gInvExposure);",
-        "pixelColor = clamp(pixelColor, 0.0, 1.0);",
-
-        "pc_fragColor = pixelColor;", // sqrt(pixelColor);',
-        //'out_FragColor = pow(pixelColor, vec4(1.0/2.2));',
-        "}",
-      ].join("\n"),
-    };
-
-    this.pathTracingGeometry = new PlaneGeometry(2, 2);
 
     // initialize texture.
     this.pathTracingUniforms.volumeTexture.value = this.volumeTexture;
     this.pathTracingUniforms.tPreviousTexture.value = this.screenTextureRenderTarget.texture;
 
-    this.pathTracingMaterial = new ShaderMaterial({
-      uniforms: this.pathTracingUniforms,
-      // defines: pathTracingDefines,
-      vertexShader: pathTracingVertexShaderSrc,
-      fragmentShader: pathTracingFragmentShaderSrc,
-      depthTest: false,
-      depthWrite: false,
-    });
-    this.pathTracingMesh = new Mesh(this.pathTracingGeometry, this.pathTracingMaterial);
-    this.pathTracingScene.add(this.pathTracingMesh);
-
-    this.screenTextureGeometry = new PlaneGeometry(2, 2);
-
-    this.screenTextureMaterial = new ShaderMaterial({
-      uniforms: this.screenTextureShader.uniforms,
-      vertexShader: this.screenTextureShader.vertexShader,
-      fragmentShader: this.screenTextureShader.fragmentShader,
-      depthWrite: false,
-      depthTest: false,
+    this.pathTracingRenderToBuffer = new RenderToBuffer(pathTracingFragmentShaderSrc, this.pathTracingUniforms);
+    this.screenTextureRenderToBuffer = new RenderToBuffer(copyImageFragShader, {
+      image: { value: this.pathTracingRenderTarget.texture },
```

Comment:
If I'm following the general move in this PR is to take a lot of the code and shaders removed from this file and have those tasks handled here by the instances of `RenderToBuffer`

Reply:
Pretty much! The objects removed from this class and replaced by these `RenderToBuffer` instances were all related to the boilerplate hoops-jumping you have to do to pretend you have an object you want to render (scene, camera, geometry, material, mesh...) when all you're really trying to tell the renderer to do is "run this shader code for every screen pixel and render the result." Now `RenderToBuffer` puts the hoops-jumping in one place instead of three.

The shader organization element is somewhat separate - I could have swapped in `RenderToBuffer` and still left the shader code it runs in big JS string arrays with `.join("\n");` at the end. But that's not nice either. if your code is instead in its own GLSL file, you can install an editor extension and edit with syntax highlighting like a real programming language.

> Seems to be running well when set to Pathtrace locally. Is there anything else you'd like for verification from reviewer?

Nope! This should be the only spot with a chance of meaningful behavior problems.